#!/usr/bin/env python
PACKAGE="vision_tasks"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

## Thresholding parameters
gen.add("low_h", int_t, 0, "Lower Bound of H", 0, 0, 255);
gen.add("high_h", int_t, 0, "Higher Bound of H", 80, 0, 255);
gen.add("low_s", int_t, 0, "Lower Bound of S", 0, 0, 255);
gen.add("high_s", int_t, 0, "Higher Bound of S", 255, 0, 255);
gen.add("low_v", int_t, 0, "Lower Bound of V", 0, 0, 255);
gen.add("high_v", int_t, 0, "Higher Bound of V", 255, 0, 255);

## Morphological parameters
gen.add("opening_mat_point", int_t, 0, "Center of the matrix for the opening operation (size extrapolated 2x+1)", 1, 1, 7);
gen.add("opening_iter", int_t, 0, "Iterations of opening applied on the thresholded image", 0, 0, 10);
gen.add("closing_mat_point", int_t, 0, "Center of the matrix for the closing operation (size extrapolated 2x+1)", 2, 1, 7);
gen.add("closing_iter", int_t, 0, "Iterations of closing applied on the opened thresholded image", 1, 0, 10);

exit(gen.generate(PACKAGE, "vision_tasks", "lineRange"));
